### PR TITLE
try to fix the NaN infinite bug

### PR DESCRIPTION
**Pull request by hjanuschka**
try to fix: https://github.com/RafalWilinski/express-status-monitor/issues/6

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "express-status-monitor",
   "version": "0.0.11",
   "description": "Realtime Monitoring for Express-based Node applications",
-  "main": "app.js",
+  "main": "index.js",
```
Comment (malixsys) [inline package.json]: 👍

File: app.js [hunk 2]
```diff
@@ -102,6 +102,19 @@ socket.on('start', function (data) {
   data[defaultSpan].responses.pop();
   data[defaultSpan].os.pop();
 
+  if(!data[defaultSpan].os[data[defaultSpan].os.length - 1]) {
+    data[defaultSpan].os[data[defaultSpan].os.length - 1]= {
```
Comment (RafalWilinski) [inline app.js]: What if `os` length will be 0? Program would try to insert something at index `-1` ;)

File: app.js [hunk 3]
```diff
@@ -102,6 +102,19 @@ socket.on('start', function (data) {
...
+      cpu:0,
+      memory:0,
+      load: [0,0,0]
+    }
+  }
+  if(!data[defaultSpan].responses[data[defaultSpan].responses.length - 1]) {
+    data[defaultSpan].responses[data[defaultSpan].responses.length - 1]= {
```
Comment (RafalWilinski) [inline app.js]: Same here.

File: app.js (hunk 2 above)
Comment (soyuka) [inline app.js]: Also, calculating the length twice is redundant. Put it in a variable at least.

File: app.js (hunk 3 above)
Comment (soyuka) [inline app.js]: again same comment

**Comment by hjanuschka** [comment]
guys i tried to stick to the coding conventions used in this project.
you are both right that it is ugly and redundant @RafalWilinski @soyuka 

i needed those fixes to get this thing working on my production app :)


indeed the right solution would be to mock the `data` object with sensitive defaults.
and/or - only return full model.

**Comment by soyuka** [comment]
> guys i tried to stick to the coding conventions used in this project.

I don't see why conventions would ask you not to put these in a variable, nor to test if the index exists.

**Comment by mattiaerre** [comment]
hi all,
maybe we can close this because of #48 ?

// @RafalWilinski @hjanuschka @malixsys @soyuka

**Pull request closed without merging**